# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/debugger/Debugger.cpp
# Commit: 31d14cf1975c
# Full Hash: 31d14cf1975c1b23ef3db176252f8cc3b31a0772
# Author: Logan Smyth <loganfsmyth@gmail.com>
# Date: 2019-12-09 09:50:39
# Regressor Bug: 1600204
# File Overlap Count: 2
# Description:
#   Bug 1600204 - Part 7: Support .this on suspended generator frames. r=jimb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D55580
# ==============================================================================

diff -r fa46d8caefde -r 31d14cf1975c js/src/debugger/Debugger.cpp
--- a/js/src/debugger/Debugger.cpp	Sun Dec 08 23:54:15 2019 +0000
+++ b/js/src/debugger/Debugger.cpp	Sun Dec 08 23:54:38 2019 +0000
@@ -1409,7 +1409,8 @@
   if (frame.debuggerNeedsCheckPrimitiveReturn()) {
     {
       AutoRealm ar(cx, frame.environmentChain());
-      if (!GetThisValueForDebuggerMaybeOptimizedOut(cx, frame, pc, thisv)) {
+      if (!GetThisValueForDebuggerFrameMaybeOptimizedOut(cx, frame, pc,
+                                                         thisv)) {
         return false;
       }
     }